pkg/web/apps/apps_http_handler_test.go

```diff
@@ -378,3 +378,53 @@ func Test_httpHandler_GetActiveAppByID(t *testing.T) {
 		})
 	}
 }
+
+func TestHTTPHandler_InitClientApp(t *testing.T) {
+	deviceWithMissingDeviceID := helpers.GetMockDevice() // See that just one mock is enough
+	deviceWithMissingDeviceID.DeviceID = ""              // See here that you can change the object to match with the criteria
+
+	deviceWithMissingVersion := helpers.GetMockDevice()
+	deviceWithMissingVersion.VersionID = ""
+
+	config := config.Get()
+	APIRoutePrefix := config.APIRoutePrefix
+	type fields struct {
+		Service Service
+	}
+	tests := []struct {
+		name        string
+		fields      fields
+		device      models.Device
+		mockService ServiceMock
+		wantErr     error
+		wantCode    int
+	}{
+		{
+			name:        "A 400 Bad Request should be returned when request body is missing device ID",
+			device:      *deviceWithMissingDeviceID,
+			mockService: *mockedService,
```

Comment:
See here the global var mockedService passed as arg.
We could also have on top of this test a specific local mockedService defined which is relevant just to this test as it is made for example into the [Test_appsService_BindingApp](https://github.com/aerogear/mobile-security-service/blob/master/pkg/web/apps/apps_service_test.go#L278) which need more scenarios than the ones globally defined in the class. In test classes, we have as standard a mockService OR mockRepo to attend the success scenarios and errors scenarios which can used for all test. See [here](https://github.com/aerogear/mobile-security-service/blob/master/pkg/web/apps/apps_service_test.go#L12) Then, if it is required a check with more specific data/result probably locally will be required a new declaration just with the impl of the func required in the test.